pcsx2/DEV9/net.cpp

```diff
@@ -82,3 +82,21 @@ void TermNet()
 		nif = nullptr;
 	}
 }
+
+NetAdapter::NetAdapter()
+{
+	//Ensure eeprom matches our default 
+	SetMACAddress(ps2MAC);
+}
+
+void NetAdapter::SetMACAddress(u8* mac)
+{
+	if (ps2MAC != mac)
```

Comment:
This will always be false. If you want to compare the actual contents of the mac's it's better to use std::array or memcmp. With preference to std::array.

Reply:
The intention is to check if the two are referring to the same region of memory (same ptr) (to avoid copying to itself)

The default constructor for NetAdapter (just above) calls SetMACAddress() with ps2MAC, this is done to ensure that the MAC address in eeprom matches what is stored in ps2MAC (as it actually isn't). Admiralty this is a little awkward

Reply:
I've tweaked it to be a little less confusing as to what I was trying to do here